src/pages/core/entrypoints/migrate.mdx

```diff
@@ -45,6 +45,42 @@ transform it.
 That's what you do in the `migrate` entrypoint. You transform the structure of
 the storage in there.
 
+## Example
+
+```rust filename="contract.rs" template="core"
+const STATE_VERSION: u64 = 1;
+
+#[entry_point]
+pub fn migrate(deps: DepsMut, _env: Env, _msg: MigrateMsg) -> StdResult<Response> {
+    // Check if the state version is olden than the current one
+    let Some(current_state_version) = deps.storage.get(b"state") else {
+        return Err(StdError::generic_err("State not found"));
+    };
+    let current_state_version: u64 = from_slice(&current_state_version)?;
+
+    if current_state_version < STATE_VERSION {
+        return Ok(Response::default());
+    }
```

Comment:
I think the recommended way to do this is using [cw2](https://github.com/CosmWasm/cw-minus/tree/main/packages/cw2), so we should probably also do that in the docs.

Reply:
Fair, yeah, gonna port that example over to using it properly